docs/learn/documentation/versioned/api/samza-sql.md

```diff
@@ -19,65 +19,70 @@ title: Samza SQL
-> ./scripts/generate-kafka-events.sh -t ProfileChangeStream -e ProfileChange
-```
+- The [group by](https://github.com/apache/samza-hello-samza/tree/master/src/main/java/samza/examples/sql/samza-sql-groupby)
+show how to do group by. Similar to Join, Group By is being actively stabilized. 
 
-#### Using the Samza SQL Console
 
+### Key Concepts
 
-The simplest SQL query is to read all events from a Kafka topic `ProfileChangeStream` and print them to the console.
+Each Samza SQL job consists of one or more Samza SQL statements.
+Each statement represents a single streaming pipeline.
 
-```bash
-> ./scripts/samza-sql-console.sh --sql "insert into log.consoleoutput select * from kafka.ProfileChangeStream"
-```
+#### SQL Representation
 
-Next, let us project a few fields from the input stream.
+Support for SQL internally uses Apache Calcite, which provides 
+SQL parsing and query planning. The query is automatically translated 
+to Samza's high level API and runs on Samza's execution engine.
 
-```bash
-> ./scripts/samza-sql-console.sh --sql "insert into log.consoleoutput select Name, OldCompany, NewCompany from kafka.ProfileChangeStream"
-```
+The mapping from SQL to the Samza's high level API is a simple 
+deterministic one-to-one mapping. For example, ```Select```, i.e., 
+projections, maps to a filter operation, while ```from``` maps to 
+a scan(s) and join(s) - if selecting from multiple streams and tables
+- operators, and so on.  
 
-You can also filter messages in the input stream based on some predicate. In this example, we filter profiles currently working at LinkedIn, whose previous employer matches the regex `.*soft`. The function `RegexMatch(regex, company)` is an example of 
-a UDF that defines a predicate. 
+The table below lists the supported SQL operations.
 
-```bash
-> ./scripts/samza-sql-console.sh --sql "insert into log.consoleoutput select Name as __key__, Name, NewCompany, RegexMatch('.*soft', OldCompany) from kafka.ProfileChangeStream where NewCompany = 'LinkedIn'"
-```
-
-
-### Running Samza SQL on YARN
-The [hello-samza](https://github.com/apache/samza-hello-samza) project has examples to get started with Samza on YARN. You can define your SQL query in a [configuration file](https://github.com/apache/samza-hello-samza/blob/master/src/main/config/pageview-filter-sql.properties) and submit it to a YARN cluster.
-
-
-```bash
-> ./deploy/samza/bin/run-app.sh --config-factory=org.apache.samza.config.factories.PropertiesConfigFactory --config-path=file://$PWD/deploy/samza/config/page-view-filter-sql.properties
-```
+ Operation | Syntax hints | Comments      
+ --- | --- | --- 
+ PROJECTION | SELECT/INSERT/UPSERT | See [SQL Grammar](#sql-grammer) below 
+ FILTERING | WHERE expression |See [SQL Grammar](#sql-grammer) below 
+ UDFs | udf_name(args)    | In both SELECT and WHERE clause 
+ JOIN | LEFT, RIGHT, FULL and NATURAL | Stream-table join. Currently not fully stable. 
+ AGGREGATION | GROUP BY | Currently not fully stable. 
```

Comment:
Only COUNT aggregate function is supported and window function is based on processing time.

Reply:
fixed